### ISSUE TITLE
from .onnx_cpp2py_export import ONNX_ML ImportError 

### ISSUE THREAD

**Issue by prabhuiitdhn** [onnx/onnx#3556]
### Help me to understand few things:

> 1. Do I need to rebuild the ONNX If I want to add any custom operator to convert one framework model to onnx model?
> 2. Do I need to rebuild the ONNX If I am debugging the ONNX code or adding/removing/replacing official code of ONNX to debug?
> 3. If in any case, If I will have to install ONNX from source what would be the best way to install ONNX from source?
> 

I have followed the following steps to install ONNX from source:
```
git clone https://github.com/onnx/onnx.git
cd onnx
git submodule update --init --recursive
# prefer lite proto
set CMAKE_ARGS=-DONNX_USE_LITE_PROTO=ON
pip install -e .

```
I have installed it but after importing I am getting an error:
>>> import onnx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/uib43225/DEFT/onnx/onnx/__init__.py", line 10, in <module>
    from .onnx_cpp2py_export import ONNX_ML
ImportError: /home/uib43225/DEFT/onnx/onnx/onnx_cpp2py_export.cpython-36m-x86_64-linux-gnu.so: undefined symbol: _ZNK6google8protobuf7Message11GetTypeNameB5cxx11Ev

python: 3.6
pytorch: 1.2.0
torchvison: 0.4.0
 

Please ask me If you need more to understand the issue. Thank you.

**Comment by jcwchen** [comment]
Hi @prabhuiitdhn,
Similar to https://github.com/onnx/onnx/issues/745. Which protobuf version did you use? Perhaps you can install protobuf    3.11.3 from source like [here](https://github.com/onnx/onnx#building-protobuf-from-source) and try again.

**Comment by prabhuiitdhn** [comment]
Hi @jcwchen:  I have installed protobuf 3.11.3 from source that you have mentioned. 

> protoc --version
> libprotoc 3.5.1

* Working on Pycharm Conda environment.

I reinstalled protobuf and onnx from source again and tried but still it did not help. Do I need to install onnx_tensorrt as dependencies?

**Comment by prabhuiitdhn** [comment]
Hi @jcwchen I have installed the protobuf v3.11.3 and onnx from source in virtual env (non-conda) through terminal and able to installed protobuf successfully.

> protoc --version
> libprotoc 3.11.3

I tried again to check the onnx model

> import onnx
> onnx.checker.check_model('/home/uib43225/DEFT/src/models/model_mot.onnx')
> print("Done.")

But got an error

>   File "test_onnx.py", line 3, in <module>
>     onnx.checker.check_model('/home/uib43225/DEFT/src/models/model_mot.onnx')
>   File "/home/uib43225/onnx_check/onnx/onnx/checker.py", line 97, in check_model
>     C.check_model_path(model)
> onnx.onnx_cpp2py_export.checker.ValidationError: checker.cc: No Op registered for _DCNv2 with domain_version of 10
> 
> ==> Context: Bad node spec for node. Name:  OpType: _DCNv2
> 

I think we are too close to fix it. Please help me to fix it. Thank you.

**Comment by askhade** [comment]
Closing this as the issue has been resolved : #3544
